addons/stock/views/report_stockpicking.xml

```diff
@@ -137,8 +162,18 @@
                             </table>
                         </div>
                     </t>
-                 </t>
              </t>
         </template>
+        <template id="report_picking">
+            <t t-foreach="docs" t-as="o">
+                <t t-if="o.picking_type_id.code == 'outgoing'">
+                    <t t-set="o" t-value="o.with_context({'lang':o.partner_id.lang})" />
+                    <t t-call="stock.report_picking_document" t-lang="o.partner_id.lang"/>
```

Comment:
i think Instead of defining new template report_picking_document, we can directly check condition on report_picking.
like : 
 +        <template id="report_picking">
+            <t t-foreach="docs" t-as="o">
                           <t t-if="o.picking_type_id.code == 'outgoing'">
                                <t t-set="o" t-value="o.with_context({'lang':o.partner_id.lang})" />
                          </t>
                     <t t-call="report.external_layout">
...
...